extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -526,19 +536,43 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
     return getCsiAmpAnalyticsConfig();
   }
 
+  /**
+   * @param {?string} autoFormat
+   * @return {boolean}
+   * @private
+   */
+  static isAutoResponsive_(autoFormat) {
+    return autoFormat === 'rspv';
+  }
+
+  /**
+   * @param {?string} autoFormat
+   * @return {boolean}
+   * @private
+   */
+  static isMCResponsive_(autoFormat) {
+    return autoFormat === 'mcrspv';
+  }
+
   /**
    * Calculates the appropriate height for a full-width responsive ad of the
    * given width.
+   * @param {string} autoFormat
    * @param {!{width: number, height: number}} viewportSize
    * @return {number}
    * @private
    */
-  static getResponsiveHeightForContext_(viewportSize) {
-    const minHeight = 100;
-    const maxHeight = Math.min(300, viewportSize.height);
-    // We aim for a 6:5 aspect ratio.
-    const idealHeight = Math.round(viewportSize.width / 1.2);
-    return clamp(idealHeight, minHeight, maxHeight);
+  static getResponsiveHeightForContext_(autoFormat, viewportSize) {
+    if (AmpAdNetworkAdsenseImpl.isAutoResponsive_(autoFormat)) {
```

Comment:
Can do same switch here (and you can make the values const) which will allow for self-documenting.  E.g.

switch (autoFormat) {
    case AUTO_RESPONSIVE_FORMAT:
       ...
    case MC_RESPONSIVE_FORMAT:
        ...
}